src/simulation/m_monopole.fpp

```diff
@@ -165,119 +180,107 @@ contains
+                                do ii = 1, num_fluids - 1
+                                    myRho = myRho + myalpha_rho(ii)
+                                    n_tait = n_tait + myalpha(ii)*gammas(ii)
+                                    B_tait = B_tait + myalpha(ii)*pi_infs(ii)
+                                end do
+                            else
+                                myRho = myalpha_rho(1)
+                                n_tait = gammas(1)
+                                B_tait = pi_infs(1)
                             end if
-                            n_tait = 1.d0/n_tait + 1.d0 !make this the usual little 'gamma'
-
-                            sound = n_tait*(q_prim_vf(E_idx)%sf(j, k, l) + ((n_tait - 1d0)/n_tait)*B_tait)/myRho
-                            sound = dsqrt(sound)
-
-                            const_sos = n_tait*(1.01d5 + ((n_tait - 1d0)/n_tait)*B_tait)/myRho
-                            const_sos = dsqrt(const_sos)
+                        else
+                            !$acc loop seq
+                            do ii = 1, num_fluids
+                                myRho = myRho + myalpha_rho(ii)
+                                n_tait = n_tait + myalpha(ii)*gammas(ii)
+                                B_tait = B_tait + myalpha(ii)*pi_infs(ii)
+                            end do
+                        end if
+                        n_tait = 1d0/n_tait + 1d0 ! The usual little 'gamma'
 
-                            term_index = 2
+                        sound = dsqrt(n_tait*(q_prim_vf(E_idx)%sf(j, k, l) + ((n_tait - 1d0)/n_tait)*B_tait)/myRho)
 
-                            angle = 0.d0
-                            angle_z = 0.d0
+                        term_index = 2
 
-                            s2 = f_g(the_time, sound, const_sos, q, term_index)* &
-                                 f_delta(j, k, l, loc_mono(:, q), length(q), q, angle, angle_z)
-                            !s2 = 1d0
+                        angle = 0d0
+                        ratio_x_r = 0d0
+                        ratio_y_r = 0d0
+                        ratio_z_r = 0d0
 
-                            if (support(q) == 5) then
-                                term_index = 1
-                                s1 = f_g(the_time, sound, const_sos, q, term_index)* &
-                                     f_delta(j, k, l, loc_mono(:, q), length(q), q, angle, angle_z)
-                            end if
+                        s2 = f_g(the_time, sound, q, term_index)* &
+                             f_delta(j, k, l, loc_mono(:, q), length(q), q, angle, ratio_x_r, ratio_y_r, ratio_z_r)
 
+                        if (support(q) == 5 .or. support(q) == 7) then
+                            term_index = 1
+                            s1 = f_g(the_time, sound, q, term_index)* &
+                                 f_delta(j, k, l, loc_mono(:, q), length(q), q, angle, ratio_x_r, ratio_y_r, ratio_z_r)
```

Comment:
much of the code in `m_monopole` never gets covered in the test suite, but perhaps easily could. check codecov

Reply:
I'll need many tests, but I'm leaving them to the end, as I'l also regenerate some of the tests

Reply:
sounds good